be/src/exprs/expr.cpp

```diff
@@ -280,7 +281,9 @@ Status Expr::create_vectorized_expr(starrocks::ObjectPool* pool, const starrocks
     }
     case TExprNodeType::CAST_EXPR: {
         if (texpr_node.__isset.child_type || texpr_node.__isset.child_type_desc) {
-            *expr = pool->add(vectorized::VectorizedCastExprFactory::from_thrift(pool, texpr_node));
+            *expr = pool->add(vectorized::VectorizedCastExprFactory::from_thrift(
+                    pool, texpr_node,
+                    (state == nullptr) ? false : state->query_options().sql_mode & MODE_ALLOC_EXCEPTION));
```

Comment:
> you could directly pass the `state->query_options().sql_mode & MODE_ALLOC_EXCEPTION` result from FE to BE, then, you needn't change a lot codes in BE

The codes in BE can't be avoided, because we must get this parameter from outer into deep level. So need change some interface, and change in many locations. It was trivial change.